sdk/azcore/internal/exported/request.go

```diff
@@ -67,6 +68,36 @@ func (ov opValues) get(value any) bool {
 	return ok
 }
 
+// NewRequestFromRequest creates a new policy.Request with an existing *http.Request
+// Exported as runtime.NewRequestFromRequest().
+func NewRequestFromRequest(req *http.Request) (*Request, error) {
+	policyReq := &Request{req: req}
+
+	if req.Body != nil {
+		// since this is an already populated http.Request we want to copy
+		// over it's body, if it has one.
+		bodyBytes, err := io.ReadAll(req.Body)
```

Comment:
If the existing body implements `io.ReadSeekCloser` then we shouldn't need to read/copy the body. We should check for that first, before making the copy.

Reply:
Fixed, test added.